packages/clerk-js/src/core/clerk.ts

```diff
@@ -665,13 +666,13 @@ export class Clerk implements ClerkInterface {
     return unsubscribe;
   };
 
-  public navigate = async (to: string | undefined): Promise<unknown> => {
+  public navigate = async (to: string | undefined, options?: NavigateOptions): Promise<unknown> => {
     if (!to || !inBrowser()) {
       return;
     }
 
     const toURL = new URL(to, window.location.href);
-    const customNavigate = this.#options.navigate;
+    const customNavigate = options?.replace && this.#options.replace ? this.#options.replace : this.#options.push;
```

Comment:
This got me a bit confused! We have `options.replace: boolean` and `this.#options.replace: () => void`. I am the only one? I would suggest we rename one of those 2 to something else.

Reply:
@dimkl I agree this is a bit confusing, but I think the usage makes more sense:

```ts
const clerk = new Clerk({ push: router.push, replace: router.replace })

// calls this.#options.push
clerk.navigate('/')

// calls this.#options.replace
clerk.navigate('/', { replace: true })
```

I'm not sure what would we rename either to considering this. 🤔

Reply:
Should we consider naming the `#options` as `historyReplace` / `historyPush`?
This suggestion comes from the comment on the `SDK-683` ticket. 
@desiprisg  Also doesn't this PR resolve that ticket?

Reply:
Will coordinate with Colin and rename.